torch_xla/csrc/ops/device_data.h

```diff
@@ -36,12 +36,10 @@ class DeviceData : public XlaNode {
   // backend data with a partitioned one in the node operands. Note that
   // this is permitted only if the node holds a placeholder.
   void Assign(std::shared_ptr<torch::lazy::BackendData> data) {
-    // TODO(yeounoh) check if the existing data is a placeholder after we
-    // address the issue where some of the sync tensors spill with device node.
     XLA_CHECK(data->shape() == data_->shape())
         << "Shape mismatch: expected (" << data_->shape().to_string()
         << "), actual (" << data->shape().to_string() << ")";
-    data_ = data;
+    data_.reset(data.get());
```

Comment:
what's the difference between these two?

Reply:
Better memory management, this is refactoring to reuse the existing data_ to hold the new data.